src/newfight.cpp

```diff
@@ -1153,11 +1181,27 @@ void hit_with_multiweapon_toggle(struct char_data *attacker, struct char_data *v
             int dam_total = convert_damage(stage(-1 * success_test(GET_BOD(attacker) + (successes == 0 ? GET_DEFENSE(attacker) : 0), 10), SERIOUS));
             
             // If the attacker dies from backlash, bail out.
-            if (damage(attacker, attacker, dam_total, TYPE_RECOIL, PHYSICAL))
+            if (damage(attacker, attacker, dam_total, TYPE_RECOIL, PHYSICAL)) {
+              
+              //Handle suprise attack/alertness here -- attacker died.
+              if (IS_NPC(def->ch) && AFF_FLAGGED(def->ch, AFF_SURPRISE))
+                AFF_FLAGS(def->ch).RemoveBit(AFF_SURPRISE);
+              if (IS_NPC(def->ch)) {
+                GET_MOBALERT(def->ch) = MALERT_CALM;
+                GET_MOBALERTTIME(def->ch) = 0;
+              }
               return;
+            }
           }
         }
       }
+      //Handle suprise attack/alertness here -- defender didn't die.
```

Comment:
If I'm reading this right, this is after the `if (!defender_died)` block-- doesn't that mean that the defender did actually die here? If so, this will crash.

Reply:
No it's at the end of that block, the defender is alive. The bracket above is closing this block: `if (successes_for_use_in_monowhip_test_check <= 0) {`

`if (!defender_died)` block is closing at line 1205 so this one's ok, the defender is definitely alive.

Reply:
Perfect, thanks!